### ISSUE TITLE
Internal Compiler Error on gossip_propagation_strategy.cpp

### ISSUE THREAD

**Issue by faizol** [hyperledger/iroha#1689]
System : Fedora 28
gcc version : 8.1.1
problem:
Internal Compiler Error on file iroha/irohad/multi_sig_transactions/impl/gossip_propagation_strategy.cpp at line/column 45:50
with error message :
internal compiler error: in is_normal_capture_proxy, at cp/lambda.c:289
                             return this->visit() | [&vec](auto e) -> bool {

I'm a noob and just started reading about Iroha. Is there a workaround I can apply to get the code compiled?

Thanks.

**Comment by l4l** [comment]
Yeah, I have exactly same issue here with gcc 8.2.0. I will take a look, when I have time one that.
I'd suggest a workaround: either downgrade your compiler (gcc6 should work) or use clang (it works for me)

**Comment by faizol** [comment]
Thanks, I got it compiled using clang.

**Comment by igor-egorov** [comment]
The symptoms say that there was not enough RAM for compiling.

**Comment by l4l** [comment]
https://www.mail-archive.com/gcc-bugs@gcc.gnu.org/msg580070.html
It's about compiler defect, actually

**Comment by l4l** [comment]
Should be fixed in #1714 now
